Assets/Scripts/InputManager.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InputManager : MonoBehaviour
+{
+    private bool draggingItem = false;
+    private GameObject draggedObject; 
+
+    Vector2 CurrentTouchPos
+    {
+        get
+        {
+            return Camera.main.ScreenToWorldPoint(Input.mousePosition);
+        }
+    }
+
+    private bool HasInput
+    {
+        get
+        {
+            return Input.GetMouseButton(0);
+        }
+    }
+
+    void Update()
+    {
+        if (HasInput)
+        {
+            PickupAndDrag();
+        }
+        else
+        {
+            if (draggingItem)
+                DropItem();
+        }
+    }
+    
+    private void PickupAndDrag()
+    {
+        var inputPos = CurrentTouchPos;
+        if (draggingItem)
+        {
+            draggedObject.transform.position = inputPos;
+        }
+        else
```

Comment:
doe anders een `return;` in de `if` statement zodat je geen else hoeft te gebruiken, dan indent de code minder en is hij meer leesbaar.